### ISSUE TITLE
Example extension is crashing Godot (4.1.1)

### ISSUE THREAD

**Issue by Stiivi** [migueldeicaza/SwiftGodot#97]
Created an extension from the README and followed the instructions. The Godot tries to load the extension and then crashes:

```
Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	       0x1a4558724 __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x1a458fc28 pthread_kill + 288
2   libsystem_c.dylib             	       0x1a449dae8 abort + 180
3   Godot                         	       0x102fc40e0 0x102be8000 + 4047072
4   libsystem_platform.dylib      	       0x1a45bea24 _sigtramp + 56
5   libSwiftGodot.dylib           	       0x13821d254 GString.init(_:) + 596 (String.swift:9)
6   libSwiftGodot.dylib           	       0x13821cff0 GString.__allocating_init(_:) + 64
7   libSwiftGodot.dylib           	       0x1382608a0 StringName.init(from:) + 136 (StringName.swift:81)
8   libSwiftGodot.dylib           	       0x138260808 StringName.__allocating_init(from:) + 64
9   libSwiftGodot.dylib           	       0x138016b54 StringName.__allocating_init(_:) + 68 (StringExtensions.swift:14)
10  libSwiftGodot.dylib           	       0x138022ce8 stripNamespace #1 <A>(_:) in register<A>(type:) + 744 (Wrapped.swift:185)
11  libSwiftGodot.dylib           	       0x1380226b0 register<A>(type:) + 872 (Wrapped.swift:205)
12  MyGame                        	       0x10d7baef4 setupScene(level:) + 112 (MyGame.swift:9)
13  libSwiftGodot.dylib           	       0x138026358 thunk for @escaping @callee_guaranteed (@unowned GDExtension.InitializationLevel) -> () + 24
14  libSwiftGodot.dylib           	       0x138025194 thunk for @escaping @callee_guaranteed (@in_guaranteed GDExtension.InitializationLevel) -> (@out ()) + 36
15  libSwiftGodot.dylib           	       0x138025044 extension_initialize(userData:l:) + 1004 (EntryPoint.swift:43)
16  libSwiftGodot.dylib           	       0x138026338 @objc extension_initialize(userData:l:) + 12
17  Godot                         	       0x106cf51e4 0x102be8000 + 68211172
18  Godot                         	       0x1030051e8 0x102be8000 + 4313576
19  Godot                         	       0x102ffdb2c 0x102be8000 + 4283180
20  Godot                         	       0x102feacbc 0x102be8000 + 4205756
21  dyld                          	       0x1a4237f28 start + 2236
```

Software information:

- Godot: 4.1.1.stable.official
- Apple Swift version 5.9-dev (LLVM 5cd14b982b25405, Swift e74109104e07cf3); Target: arm64-apple-macosx13.0

Package.swift:

```
// swift-tools-version: 5.8
// The swift-tools-version declares the minimum version of Swift required to build this package.

import PackageDescription

let package = Package(
    name: "MyGame",
    platforms: [
        .macOS(.v10_15),
        .iOS ("16.0")
    ],
    products: [
        // Products define the executables and libraries a package produces, making them visible to other packages.
        .library(
            name: "MyGame",
            type: .dynamic,
            targets: ["MyGame"]),
    ],
    dependencies: [
        .package(url: "https://github.com/migueldeicaza/SwiftGodot", branch: "main")
    ],
    targets: [
        // Targets are the basic building blocks of a package, defining a module or a test suite.
        // Targets can depend on other targets in this package and products from dependencies.
        .target(
            name: "MyGame",
            dependencies: ["SwiftGodot"],
            swiftSettings: [.unsafeFlags(["-suppress-warnings"])],
            linkerSettings: [.unsafeFlags(
                ["-Xlinker", "-undefined",
                 "-Xlinker", "dynamic_lookup"]
            )]

        ),
    ]
)
```

**Comment by migueldeicaza** [comment]
Please try 4.0, as 4.1 changed the calling conventions for calling methods (there is a bug with some details)

**Comment by migueldeicaza** [comment]
New PR is up, would love if you could try it:

https://github.com/migueldeicaza/SwiftGodot/pull/104

**Comment by migueldeicaza** [comment]
Fixed in the 4.1 branch
